sdk/core/core-lro/src/lroEngine/requestUtils.ts

```diff
@@ -35,23 +35,21 @@ export function inferLroMode(
   requestMethod: string,
   rawResponse: RawResponse
 ): LroConfig {
-  if (getAzureAsyncOperation(rawResponse) !== undefined) {
+  const hasAzureAsync = getAzureAsyncOperation(rawResponse) !== undefined;
+  const hasLocation = getLocation(rawResponse) !== undefined;
+  const hasOpLocation = getOperationLocation(rawResponse) !== undefined;
+  if (hasAzureAsync || hasLocation || hasOpLocation) {
     return {
-      mode: "AzureAsync",
+      mode: "Location",
       resourceLocation:
-        requestMethod === "PUT"
-          ? requestPath
-          : requestMethod === "POST" || requestMethod === "PATCH"
-          ? getLocation(rawResponse)
+        hasAzureAsync || hasOpLocation
```

Comment:
nit: triple-ternary operators!  Does the value of `hasLocation` matter at all here?

Reply:
In the case of a post or a patch, we will need the value in the `Location` header if it exists, see line 49. `hasLocation` itself does not matter at this point.

Reply:
Ok I created `findResourceLocation` to simplify the conditional.

Reply:
if `hasAzureAsync || hasOpLocation` is true, the value of `hasLocation` doesn't matter since we don't use it.  If `hasAzureAsync || hasOpLocation` is false, `hasLocation` must be true and we return `{ mode: "Location", resourceLocation: undefined }`?

Reply:
if `hasAzureAsync || hasOpLocation` is false then `hasLocation` must be true here, but we return undefined for location?

Reply:
because it has a different meaning then and we do not care about it.